internal/operands/tekton-tasks/tekton-tasks.go

```diff
@@ -0,0 +1,313 @@
+	executeInVMTaskName:          environment.GetCleanupVMImage,
+	modifyWindowsVMIsoFileName:   environment.GetCleanupVMImage,
+}
+
+func init() {
+	utilruntime.Must(pipeline.AddToScheme(common.Scheme))
+}
+
+type tektonTasks struct {
+	tasks           []pipeline.Task
+	serviceAccounts []v1.ServiceAccount
+	roleBindings    []rbac.RoleBinding
+	clusterRoles    []rbac.ClusterRole
+}
+
+var _ operands.Operand = &tektonTasks{}
+
+func New(bundle *tektonbundle.Bundle) *tektonTasks {
+	tt := &tektonTasks{
+		tasks:           bundle.Tasks,
+		serviceAccounts: bundle.ServiceAccounts,
+		roleBindings:    bundle.RoleBindings,
+		clusterRoles:    bundle.ClusterRoles,
+	}
+
+	tt.filterUnusedObjects()
+
+	return tt
+}
+
+func (t *tektonTasks) Name() string {
+	return operandName
+}
+
+func (t *tektonTasks) WatchClusterTypes() []client.Object {
+	return []client.Object{
+		&rbac.ClusterRole{},
+		&pipeline.ClusterTask{},
+		&rbac.RoleBinding{},
+		&v1.ServiceAccount{},
+	}
+}
+
+func (t *tektonTasks) WatchTypes() []client.Object {
+	return nil
+}
+
+func (t *tektonTasks) RequiredCrds() []string {
+	return requiredCRDs
+}
+
+func (t *tektonTasks) filterUnusedObjects() {
+	newTasks := []pipeline.Task{}
+	for _, task := range t.tasks {
+		if _, ok := AllowedTasks[task.Name]; ok {
+			newTasks = append(newTasks, task)
+		}
+	}
+	t.tasks = newTasks
+
+	newSA := []v1.ServiceAccount{}
+	for _, sa := range t.serviceAccounts {
+		if _, ok := AllowedTasks[strings.TrimSuffix(sa.Name, "-task")]; ok {
+			newSA = append(newSA, sa)
+		}
+	}
+	t.serviceAccounts = newSA
+
+	newRB := []rbac.RoleBinding{}
+	for _, rb := range t.roleBindings {
+		if _, ok := AllowedTasks[strings.TrimSuffix(rb.Name, "-task")]; ok {
+			newRB = append(newRB, rb)
+		}
+	}
+	t.roleBindings = newRB
+
+	newCR := []rbac.ClusterRole{}
+	for _, cr := range t.clusterRoles {
+		if _, ok := AllowedTasks[strings.TrimSuffix(cr.Name, "-task")]; ok {
+			newCR = append(newCR, cr)
+		}
+	}
+	t.clusterRoles = newCR
+}
+
+func (t *tektonTasks) Reconcile(request *common.Request) ([]common.ReconcileResult, error) {
+	var results []common.ReconcileResult
+	var reconcileFunc []common.ReconcileFunc
+	reconcileFunc = append(reconcileFunc, reconcileTektonTasksFuncs(t.tasks)...)
+	reconcileFunc = append(reconcileFunc, reconcileClusterRoleFuncs(t.clusterRoles)...)
+	reconcileFunc = append(reconcileFunc, reconcileServiceAccountsFuncs(t.serviceAccounts)...)
+	reconcileFunc = append(reconcileFunc, reconcileRoleBindingFuncs(t.roleBindings)...)
+
+	reconcileTektonBundleResults, err := common.CollectResourceStatus(request, reconcileFunc...)
+	if err != nil {
+		return nil, err
+	}
+
+	upgradingNow := isUpgradingNow(request)
+	for _, r := range reconcileTektonBundleResults {
+		if !upgradingNow && (r.OperationResult == common.OperationResultUpdated) {
+			request.Logger.Info(fmt.Sprintf("Changes reverted in tekton tasks: %s", r.Resource.GetName()))
+		}
+	}
+	return append(results, reconcileTektonBundleResults...), nil
```

Comment:
`codes style` - It's unnecessary to use `append()`, you can return `reconcileTektonBundleResults` directly.

Reply:
You're right, done.